CONTRIBUTING.md

```diff
@@ -42,6 +42,10 @@ To quickly get up and running, you can use the `scripts/eachdist.py` tool that
 ships with this project. First create a virtualenv and activate it.
 Then run `python scripts/eachdist.py develop` to install all required packages
 as well as the project's packages themselves (in `--editable` mode).
+
+Further, you'll want to clone the Contrib repo locally to resolve paths needed
```

Comment:
@ocelotl I've added a comment here in `CONTRIBUTING.md` so that people know they have to clone the Contrib repo. Also it is in the `test.yml` github actions workflow.

However I agree it's not obvious. A great solution would be to add it to the `scripts/eachdist.py develop` command like @toumorokoshi suggested, we can create an issue and follow up with that later if that's alright with you?

Reply:
Shouldn't we add it to [`comments_pre`](https://tox.readthedocs.io/en/latest/config.html#conf-commands_pre) in `tox.ini` instead?

Reply:
I don't think we should because you don't need to run `git clone` every time you want to run `tox -e lint`. You just want to run it _once_ before.

Further your "clone" can get out of date easily if someone merges a PRs that requires the "clone" to be on a new commit SHA. That's why I think a script that knows how to update the commit the cloned repo HEAD points to would be the nicest developer workflow.

This PR is just to get the Contrib tests in, but a great quick follow up would be to improve this experience.

Reply:
Is it possible to install the instrumentations as packages in the corresponding testing virtual environment instead of having to clone the repo?

Reply:
Yes... but it's very ugly 😓 

On the [pip documentation](https://pip.pypa.io/en/stable/reference/pip_install/#git) there is support for installing from git, but it assumes only 1 package for every git repository. You _can_ tell it to go into a `subdirectory` to find a package, but it will still clone the _entire_ repo just to download 1 package. And we would have to do that for every package.

Cloning the Contrib repo also has the benefit of getting the `opentelemetry-python-contrib/tox.ini` file so that we can just run it directly, instead of polluting our core `tox.ini` file with references to the Contrib repo packages.